datafusion/optimizer/src/utils.rs

```diff
@@ -101,9 +101,7 @@ pub(crate) fn replace_qualified_name(
 ) -> Result<Expr> {
     let alias_cols: Vec<Column> = cols
         .iter()
-        .map(|col| {
-            Column::from_qualified_name(format!("{}.{}", subquery_alias, col.name))
-        })
+        .map(|col| Column::new(Some(subquery_alias), col.name()))
```

Comment:
in upstream this was changed the same way in https://github.com/apache/datafusion/pull/12443
can we backport that PR along with tests?

benefits:

- we would have tests. 
- when upgrading to newer DF version, we would better know what to do with this change, in case it conflicts (it won't conflict today, but it could confict if upstream code diverges further)

Reply:
Oh yes, that's good call thanks! I wasn't tracking upstream, lol